File: CS/Scaffolding.DetailCollections.EFCore/Migrations/20161006093701_MyFirstMigration.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore.Migrations;
using Microsoft.EntityFrameworkCore.Metadata;

namespace Scaffolding.DetailCollections.EFCore.Migrations {
    public partial class MyFirstMigration : Migration {
        protected override void Up(MigrationBuilder migrationBuilder) {
            migrationBuilder.CreateTable(
                name: "Departments",
                columns: table => new
                {
                    DepartmentID = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    Name = table.Column<string>(nullable: true)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_Departments", x => x.DepartmentID);
                });

            migrationBuilder.CreateTable(
                name: "Courses",
                columns: table => new
                {
                    CourseID = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    DepartmentID = table.Column<int>(nullable: false),
                    Title = table.Column<string>(nullable: false)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_Courses", x => x.CourseID);
                    table.ForeignKey(
                        name: "FK_Courses_Departments_DepartmentID",
                        column: x => x.DepartmentID,
                        principalTable: "Departments",
                        principalColumn: "DepartmentID",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateTable(
                name: "Employees",
                columns: table => new
                {
                    EmployeeID = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    DepartmentID = table.Column<int>(nullable: false),
                    Name = table.Column<string>(nullable: true)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_Employees", x => x.EmployeeID);
                    table.ForeignKey(
                        name: "FK_Employees_Departments_DepartmentID",
                        column: x => x.DepartmentID,
                        principalTable: "Departments",
                        principalColumn: "DepartmentID",
                        onDelete: ReferentialAction.Cascade);
                });

            migrationBuilder.CreateIndex(
                name: "IX_Courses_DepartmentID",
                table: "Courses",
                column: "DepartmentID");

            migrationBuilder.CreateIndex(
                name: "IX_Employees_DepartmentID",
                table: "Employees",
                column: "DepartmentID");
        }

        protected override void Down(MigrationBuilder migrationBuilder) {
            migrationBuilder.DropTable(
                name: "Courses");

            migrationBuilder.DropTable(
                name: "Employees");

            migrationBuilder.DropTable(
                name: "Departments");
        }
    }
}

File: VB/Scaffolding.DetailCollections.EFCore/Migrations/20161006093701_MyFirstMigration.vb
Imports System
Imports System.Collections.Generic
Imports Microsoft.EntityFrameworkCore.Migrations
Imports Microsoft.EntityFrameworkCore.Metadata

Namespace Scaffolding.DetailCollections.EFCore.Migrations
	Partial Public Class MyFirstMigration
		Inherits Migration

		Protected Overrides Sub Up(ByVal migrationBuilder As MigrationBuilder)
			migrationBuilder.CreateTable(name:= "Departments", columns:= Function(table) New With {Key .DepartmentID = table.Column(Of Integer)(nullable:= False).Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn), Key .Name = table.Column(Of String)(nullable:= True)}, constraints:= Function(table) table.PrimaryKey("PK_Departments", Function(x) x.DepartmentID))

			migrationBuilder.CreateTable(name:= "Courses", columns:= Function(table) New With {Key .CourseID = table.Column(Of Integer)(nullable:= False).Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn), Key .DepartmentID = table.Column(Of Integer)(nullable:= False), Key .Title = table.Column(Of String)(nullable:= False)}, constraints:= Sub(table)
					table.PrimaryKey("PK_Courses", Function(x) x.CourseID)
					table.ForeignKey(name:= "FK_Courses_Departments_DepartmentID", column:= Function(x) x.DepartmentID, principalTable:= "Departments", principalColumn:= "DepartmentID", onDelete:= ReferentialAction.Cascade)
				End Sub)

			migrationBuilder.CreateTable(name:= "Employees", columns:= Function(table) New With {Key .EmployeeID = table.Column(Of Integer)(nullable:= False).Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn), Key .DepartmentID = table.Column(Of Integer)(nullable:= False), Key .Name = table.Column(Of String)(nullable:= True)}, constraints:= Sub(table)
					table.PrimaryKey("PK_Employees", Function(x) x.EmployeeID)
					table.ForeignKey(name:= "FK_Employees_Departments_DepartmentID", column:= Function(x) x.DepartmentID, principalTable:= "Departments", principalColumn:= "DepartmentID", onDelete:= ReferentialAction.Cascade)
				End Sub)

			migrationBuilder.CreateIndex(name:= "IX_Courses_DepartmentID", table:= "Courses", column:= "DepartmentID")

			migrationBuilder.CreateIndex(name:= "IX_Employees_DepartmentID", table:= "Employees", column:= "DepartmentID")
		End Sub

		Protected Overrides Sub Down(ByVal migrationBuilder As MigrationBuilder)
			migrationBuilder.DropTable(name:= "Courses")

			migrationBuilder.DropTable(name:= "Employees")

			migrationBuilder.DropTable(name:= "Departments")
		End Sub
	End Class
End Namespace
